src/librepgp/stream-armor.cpp

```diff
@@ -193,9 +194,9 @@ armor_read_trailer(pgp_source_t *src)
     }
 
     stlen = strlen(param->armorhdr);
-    strncpy(st, ST_ARMOR_END, 8); /* 8 here is mandatory */
+    memcpy(st, ST_ARMOR_END, std::min<size_t>(sizeof(ST_ARMOR_END)-1, 8)); /* 8 here is mandatory */
```

Comment:
I think instead of std::min usage it would more readable/safe to check whether stlen + 8 + 1 less or equal to sizeof(st), and then use memcpy().